### DISCUSSION TITLE
About "file size" issue, file size should change to uint64？

### DISCUSSION THREAD

**Discussion by MoonLord-LM** [libgit2/libgit2#6755] [Q&A] [answered]
https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getfilesizeex#parameters
https://github.com/libgit2/libgit2/blob/main/include/git2/index.h#L48-L51

Like this code, "uint32_t file_size" should change to uint64.
The max value of uint32 file size is 4GB, it is not big enough.

**Comment by MoonLord-LM** [comment]
https://gitlab.com/tortoisegit/tortoisegit/-/issues/4108

**Comment by ethomson** [comment, marked as answer]
This is a git file structure, we cannot change it without breaking compatibility with git itself.

This part of the index is known as the “cache” and contains information about the files on disk to avoid unnecessary reads. The data can be lower fidelity - and doesn’t need full precision.
